pmd-core/src/main/java/net/sourceforge/pmd/Report.java

```diff
@@ -512,7 +522,7 @@ public long getElapsedTimeInMillis() {
         return end - start;
     }
 
-    public List<SynchronizedReportListener> getSynchronizedListeners() {
+    public List<ThreadSafeReportListener> getSynchronizedListeners() {
```

Comment:
I'd say, rename the method right away.
I think, it's ok to break the API since we are on the master (5.6.x) branch only.
However, we should have a short notice in the **changelog** under section "API Changes" - just in case.
I assume, integration/upgrading of PMD with other tools, usually works like this: Upgrade PMD dependency and if it compiles and tests successfully -> great. If not, maybe look at the changelog or just read the code...